### PR TITLE
SDI-603: ✨ Physical details endpoint

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/PhysicalDetailService.kt [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+  private fun validateDetailRequest(detailRequest: PhysicalDetailRequest): Unit = with(detailRequest) {
+    if (prisonIds.isNullOrEmpty()) {
+      throw BadRequestException("Invalid physical detail search  - please provide prison locations to filter by")
+    }
+    if ((minHeight ?: 0) < 0) {
+      throw BadRequestException("Minimum height cannot be less than 0")
+    }
+    if ((maxHeight ?: 0) < 0) {
+      throw BadRequestException("Minimum height cannot be less than 0")
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/PhysicalDetailService.kt:47]: Typo on minimum

Comment (petergphillips) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/PhysicalDetailService.kt:47]: took me a while here to understand your comment - I was thinking that I'd spelt minimum incorrectly 😂

Comment (petergphillips) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/PhysicalDetailService.kt:47]: will correct on next PR if that's okay - got to add height etc.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PhysicalDetailResourceTest.kt [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+  private companion object {
+    private var initialiseSearchData = true
+  }
+
+  @BeforeEach
+  fun setup() {
+    if (initialiseSearchData) {
+
+      setupIndexes()
+      indexPrisoners()
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PhysicalDetailResourceTest.kt:22]: Can this extend `AbstractSearchDataIntegrationTest` from the last PR?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PhysicalDetailResourceTest.kt [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    webTestClient.post().uri("/physical-detail")
+      .body(BodyInserters.fromValue(gson.toJson(PhysicalDetailRequest(minHeight = 100, prisonIds = listOf("MDI")))))
+      .headers(setAuthorisation(roles = listOf("ROLE_PRISONER_SEARCH")))
+      .header("Content-Type", "application/json")
+      .exchange()
+      .expectStatus().isOk
+  }
+
+  @Test
+  fun `can perform a detail search for ROLE_GLOBAL_SEARCH and ROLE_PRISONER_SEARCH role`() {
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PhysicalDetailResourceTest.kt:111]: Not really sure what this is testing?

**Review by mikehalmamoj — APPROVED** [review]

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PhysicalDetailResourceTest.kt (hunk 2 above)
Comment (petergphillips) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PhysicalDetailResourceTest.kt:22]: certainly, will do in next PR.  Raised them both at the same time and didn't want one of them conflicting with the other one

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PhysicalDetailResourceTest.kt (hunk 3 above)
Comment (petergphillips) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PhysicalDetailResourceTest.kt:111]: good question, it was a copy and paste from the PrisonerDetailResourceTest.  Will investigate and fix in next PR.

Comment (andymarke) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PhysicalDetailResourceTest.kt:111]: Isn't testing if they have both roles it still works ? Which is OTT

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PhysicalDetailResourceTest.kt [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+      .headers(setAuthorisation(roles = listOf("ROLE_GLOBAL_SEARCH", "ROLE_PRISONER_SEARCH")))
+      .header("Content-Type", "application/json")
+      .exchange()
+      .expectStatus().isOk
+  }
+
+  @Test
+  fun `will page the results - first page limited to size`(): Unit = physicalDetailSearch(
+    detailRequest = PhysicalDetailRequest(minHeight = 100, prisonIds = listOf("MDI", "LEI"), pagination = PaginationRequest(0, 2)),
+    expectedPrisoners = listOf("A1090AA", "A7089EY"),
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PhysicalDetailResourceTest.kt:123]: Where are all these prisoners defined? Feels like they should be in the test

**Pull request merged**